src/hex.py

```diff
@@ -14,9 +14,19 @@
     case "encode":
         # Implement the encoding here
         encoding = ""
+        y = []
+        for i in x:
+            y.append(hex(ord(i)))
+        encoding = ''.join(y)
```

Comment:
Good solution. However, there is a tradition in programming that we use `i` and `j` as variables that contain integers. This isn't universal, but so close to that people in general will find it confusing when you use `i` as a loop variable over characters.

Another way to implement the loop is through a "list comprehension" or a "generator expression". It looks a little more daunting until you get used to them, but you can write very succinct code:

```python
encoding = ''.join(hex(ord(c)) for c in x)
```

Here, the `(hex(ord(c)) for c in x)` is a generator expression, if you want a list comprehension it is `[hex(ord(c)) for c in x]`. It is just a short way of writing the loop with an append that you have.